File: backend/KioskAPI/Models/SystemInfo.cs
namespace KioskAPI.Models;

public class SystemInfo
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string MachineName { get; set; } = string.Empty;
    public string Hostname { get; set; } = string.Empty;
    public string OsVersion { get; set; } = string.Empty;
    public string IpAddress { get; set; } = string.Empty;
    public string? Uptime { get; set; }
    public string? CurrentUser { get; set; }
    public DateTime CollectedAt { get; set; } = DateTime.UtcNow;
}

File: frontend/kiosk-ui/src/types/systemInfo.ts
export interface SystemInfo {
  id: string;
  machineName: string;
  hostname: string;
  osVersion: string;
  ipAddress: string | null;
  uptime: string | null;
  currentUser: string | null;
  cpuCores: number | null;
  totalRamMb: number | null;
  freeRamMb: number | null;
  diskSummary: string | null;
  lastBoot: string | null;
  collectedAt: string;
}

export interface EventLogEntry {
  id: string;
  machineName: string;
  logName: string;
  source: string;
  eventId: number;
  level: string;
  message: string;
  timestamp: string;
  collectedAt: string;
}

export interface EventLogFilter {
  logName: string;
  level: string;
  hoursBack: number;
  maxResults: number;
}
